lib/server.rb

```diff
@@ -21,25 +21,55 @@ def reset_count
     @count = -1
   end
 
-  def format_response(request_lines)
-    reset_count if request_lines[0].include? "/clear_count"
+  def format_response(normalized_lines)
+    response = "#{diagnostic_template(normalized_lines)}"
+    reset_count if normalized_lines[1][1].include? "/clear_count"
 
-    response = "<p>Hello, World! (#{count})</p>"
+    # response = "<p>Hello, World! (#{count})</p>"
     output = "<html><head></head><body>#{response}</body></html>"
     headers = ["http/1.1 200 ok",
-               "date: #{Time.zone.now.strftime('%a, %e %b %Y %H:%M:%S %z')}",
+               "date: #{Time.now.strftime('%a, %e %b %Y %H:%M:%S %z')}",
                "server: ruby",
                "content-type: text/html; charset=iso-8859-1",
                "content-length: #{output.length}\r\n\r\n"].join("\r\n")
     send_response(headers, output)
   end
+  #
+  # def format_lines(request_lines)
+  #   request_lines.map { |e| e.split(" ") }.sort
+  # end
+
+  def diagnostic_template(normalized_lines)
+    # formatted_lines = format_lines(request_lines)
+    "<pre>" +
+    "Verb: #{normalized_lines[1][0]}\n" +
+    "Path: #{normalized_lines[1][1]}\n" +
+    "Protocol: #{normalized_lines[1][2]}\n" +
+    "Host: #{normalized_lines.last[1].split(":")[0]}\n" +
+    "Port: #{normalized_lines.last[1].split(":")[1]}\n" +
+    "Origin: #{normalized_lines.last[1].split(":")[0]}\n" +
```

Comment:
Use `\` instead of `+` or `<<` to concatenate those strings.<br>Use 2 (not 0) spaces for indenting an expression spanning multiple lines.<br>Prefer single-quoted strings inside interpolations.